src/zui/components/ZUISection/index.tsx

```diff
@@ -0,0 +1,369 @@
+import { Box, Typography } from '@mui/material';
+import { FC } from 'react';
+
+import ZUIDivider from '../ZUIDivider';
+import { ZUIOrientation } from '../types';
+
+type SubSectionBase = {
+  /**
+   * The content of the sub section.
+   */
+  renderContent: () => JSX.Element;
+
+  /**
+   * The subtitle of the sub section.
+   */
+  subtitle?: string;
+
+  /**
+   * The title of the sub section,
+   */
+  title: string;
```

Comment:
Very often, these will be messages. We can use `useMessages()` to get a string, but I just had the idea that what if instead of typing this as `string`, we typed it as `string | MsgProps` which would allow the value to be, for example:

```ts
title={{
  id: messageIds.some.message.id,
  values: {
    name: 'Some value',
  }
}}
```

I'm absolutely not suggesting that we do this now, but it's an idea that came to me now and I wanted to just share in order to not be alone with it. 😊

Reply:
![images](https://github.com/user-attachments/assets/c5d16f6f-74f7-4312-809b-b42c22c34bd4)